pkg/planner/core/plan_cache.go

```diff
@@ -618,6 +663,28 @@ func rebuildRange(p Plan) error {
 	return nil
 }
 
+// reEvaluateParamMarkerConst updates all constants to their current ParamMarkers
+func reEvaluateParamMarkerConst(ctx PlanContext, exprs []expression.Expression) error {
```

Comment:
I can understand what this function is doing, but I'm not sure whether this operation is necessary.
It seems like we don't need to update `Constant.Value` manually, since when using or evaluating a `Constant`, it first checks whether it is a `Parameter` automatically:
<img width="1528" alt="image" src="https://github.com/pingcap/tidb/assets/7499936/27f94618-c0e9-4a90-99bb-faeb5de5a98b">

What if we remove these `reEvaluateParamMakerConst` operations, will it cause any error?

Reply:
Ah, now I see, since I replaced the old 'quick' [Batch]PointGet limited pruning, GetPhysID(), which used datum.GetInt64(), with the generic partitioned_table.GetPartition[Idx]ByRow() which uses Eval, I can now remove this.

I will remove this from the PR.